Shopify/cli
# Themes cli migration cli kit

### WHY are these changes introduced?
Folow-up: #1125 


CLI2 is going to be deprecated in some weeks. Although the team is working on the migration of the `themes` commands from Ruby to Typescript, a fallback is needed in case those task don't finish before the deadline. 

Another disadvantage of having both CLI version on different repositories is that each time a new feature of fix is needed, two PRs must be created and the deployment process should be synced between them.

So with this PR the CLI2 is going to be embedded inside the CLI3. As a first step the complete CLI2 will be copied inside but in a next step only the code related with `theme` commands should be retained


### WHAT is this pull request doing?
- Move the Ruby CLI2 codebase to the `cli-kit` module. Copied inside `assets/cli-ruby` so it's automatically included in the generated package
- Add a feature flag to switch between the using of the `embedded` version and the `bundled` one
- Create a new github workflow to execute `lint` and `test` task for the embedded code.



### MISSING points
- `assets/cli-source` includes the complete CLI codebase. Retain only `theme commands` related code.

### How to test your changes?
- Run any theme command using `--verbose` flag and check the exec message of the CLI2. It should be similar to this
```
Running system process:
  · Command: /Users/alvarogutierrez/src/github.com/Shopify/cli/packages/cli-kit/assets/cli-sources/bin/shopify theme check /Users/alvarogutierrez/cli-themes/first-theme --output text
  · Working directory: /Users/alvarogutierrez/cli-themes/first-theme
```
- In the release version the `bundled` version of the CLI2 is going to be used so far. To simulate this in the local environment run any `theme` command prepending this `SHOPIFY_CLI_BUNDLED_THEME_CLI=1`


### Additional notes
In case you want to run lint  and test tasks locally you should do it manually running either `pnpm lint:ruby` or `pnpm test:ruby`  inside the `cli-kit` folder

### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
